bridge/src/main/java/com/bitwarden/bridge/util/EncryptionUtils.kt

```diff
@@ -0,0 +1,195 @@
+package com.bitwarden.bridge.util
+
+import com.bitwarden.bridge.IBridgeService
+import com.bitwarden.bridge.IBridgeServiceCallback
+import com.bitwarden.bridge.model.AddTotpLoginItemData
+import com.bitwarden.bridge.model.AddTotpLoginItemDataJson
+import com.bitwarden.bridge.model.EncryptedAddTotpLoginItemData
+import com.bitwarden.bridge.model.EncryptedSharedAccountData
+import com.bitwarden.bridge.model.SharedAccountData
+import com.bitwarden.bridge.model.SharedAccountDataJson
+import com.bitwarden.bridge.model.SymmetricEncryptionKeyData
+import com.bitwarden.bridge.model.toByteArrayContainer
+import kotlinx.serialization.encodeToString
+import java.security.MessageDigest
+import java.security.SecureRandom
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+/**
+ * Generate a symmetric [SecretKey] that will used for encrypting IPC traffic.
+ *
+ * This is intended to be used for implementing [IBridgeService.getSymmetricEncryptionKeyData].
+ */
+fun generateSecretKey(): Result<SecretKey> = runCatching {
+    val keygen = KeyGenerator.getInstance("AES")
```

Comment:
Can we look into using `KeyProperties.KEY_ALGORITHM_AES` etc. for these? When creating the Cipher we can probably use something like this from the biometrics code:

```kotlin
private const val CIPHER_TRANSFORMATION =
    KeyProperties.KEY_ALGORITHM_AES + "/" +
        KeyProperties.BLOCK_MODE_CBC + "/" +
        KeyProperties.ENCRYPTION_PADDING_PKCS7
```

Reply:
So I was able to remove a few of the raw strings here. 

I still hit a sticky wicket with `KeyProperties.ENCRYPTION_PADDING_PKCS7` blowing up test because that algo is not available in our test environment. 

I failed to mock out that constant. To my knowledge and reading the [mockk docs](https://mockk.io/), I don't think mockk supports mocking a static constant. It can mock top level constants and extension functions, but I was't able to make it work for this static class constant.

Reply:
OK sounds good. Thanks for that context!